node/src/components/api_server/http_server.rs

```diff
@@ -29,31 +30,35 @@ pub(super) async fn run<REv: ReactorEventT>(
     effect_builder: EffectBuilder<REv>,
     mut data_receiver: mpsc::UnboundedReceiver<SseData>,
 ) {
+    // REST filters.
+    let rest_status = rest_server::create_status_filter(effect_builder);
+    let rest_metrics = rest_server::create_metrics_filter(effect_builder);
+
     // RPC filters.
-    let put_deploy = rpcs::account::PutDeploy::create_filter(effect_builder);
-    let get_block = rpcs::chain::GetBlock::create_filter(effect_builder);
-    let get_global_state_hash = rpcs::chain::GetGlobalStateHash::create_filter(effect_builder);
-    let get_item = rpcs::state::GetItem::create_filter(effect_builder);
-    let get_balance = rpcs::state::GetBalance::create_filter(effect_builder);
-    let get_deploy = rpcs::info::GetDeploy::create_filter(effect_builder);
-    let get_peers = rpcs::info::GetPeers::create_filter(effect_builder);
-    let get_status = rpcs::info::GetStatus::create_filter(effect_builder);
-    let get_metrics = rpcs::info::GetMetrics::create_filter(effect_builder);
+    let rpc_put_deploy = rpcs::account::PutDeploy::create_filter(effect_builder);
+    let rpc_get_block = rpcs::chain::GetBlock::create_filter(effect_builder);
+    let rpc_get_global_state_hash = rpcs::chain::GetGlobalStateHash::create_filter(effect_builder);
+    let rpc_get_item = rpcs::state::GetItem::create_filter(effect_builder);
+    let rpc_get_balance = rpcs::state::GetBalance::create_filter(effect_builder);
+    let rpc_get_deploy = rpcs::info::GetDeploy::create_filter(effect_builder);
+    let rpc_get_peers = rpcs::info::GetPeers::create_filter(effect_builder);
+    let rpc_get_status = rpcs::info::GetStatus::create_filter(effect_builder);
 
     // Event stream channels and filter.
     let (broadcaster, mut new_subscriber_info_receiver, sse_filter) =
         sse_server::create_channels_and_filter();
 
     let service = warp_json_rpc::service(
-        put_deploy
-            .or(get_block)
-            .or(get_global_state_hash)
-            .or(get_item)
-            .or(get_balance)
-            .or(get_deploy)
-            .or(get_peers)
-            .or(get_status)
-            .or(get_metrics)
+        rest_status
+            .or(rest_metrics)
+            .or(rpc_put_deploy)
```

Comment:
Why are REST metrics argument to `warp_json_rpc`?

Reply:
I have to use to use that service if I want any JSON-RPC handling via warp at all.  I had a brief look, and it's only creating a very thin wrapper around the default warp service (to handle caching RPC requests I believe).  The event-stream filters are tacked on to the end of the filters too.

I did try to avoid this, but it seems like I either would need to have a different service for each, which requires a different listening socket for each (not as we agreed in the CEP), or we move away from the warp collection of crates and go more low-level, probably replicating a lot of the warp code in our own codebase.

Reply:
Thanks for explaining. It's weird that we're wrapping REST endpoints with JSON-RPC services though.